src/thunderbots/software/test/ai/hl/stp/evaluation/robot.cpp

```diff
@@ -87,3 +87,83 @@ TEST(RobotEvaluationTest,
     EXPECT_FALSE(Evaluation::robotOrientationWithinAngleThresholdOfTarget(
         position, orientation, target, threshold));
 }
+
+TEST(RobotEvaluationTest, has_possession_directly_in_front_of_robot)
+{
+    Point ball_position  = Point(0.07, 0);
+    Vector ball_velocity = Vector(0, 0);
+    Timestamp timestamp  = Timestamp::fromSeconds(0);
+    Ball ball            = Ball(ball_position, ball_velocity, timestamp);
+
+    Robot robot = Robot(0, Point(0, 0), Vector(), Angle::zero(), AngularVelocity::zero(),
+                        timestamp);
+
+    EXPECT_TRUE(Evaluation::robotHasPossession(ball, robot));
+}
+
+TEST(RobotEvaluationTest, has_possession_directly_in_front_of_robot_but_robot_turned)
+{
+    Point ball_position  = Point(0.07, 0);
+    Vector ball_velocity = Vector(0, 0);
+    Timestamp timestamp  = Timestamp::fromSeconds(0);
+    Ball ball            = Ball(ball_position, ball_velocity, timestamp);
+
+    Robot robot = Robot(0, Point(0, 0), Vector(), Angle::half(), AngularVelocity::zero(),
+                        timestamp);
+
+    EXPECT_FALSE(Evaluation::robotHasPossession(ball, robot));
+}
+
+TEST(RobotEvaluationTest, has_possession_robot_moving)
+{
+    Point ball_position  = Point(0.07, 0);
+    Vector ball_velocity = Vector(0, 0);
+    Timestamp timestamp  = Timestamp::fromSeconds(0);
+    Ball ball            = Ball(ball_position, ball_velocity, timestamp);
+
+    Robot robot = Robot(0, Point(0, 0), Vector(1, 1), Angle::zero(),
+                        AngularVelocity::zero(), timestamp);
+
+    EXPECT_TRUE(Evaluation::robotHasPossession(ball, robot));
+}
+
+TEST(RobotEvaluationTest, has_possession_ball_far_away)
+{
+    Point ball_position  = Point(-1, -2);
+    Vector ball_velocity = Vector(0, 0);
+    Timestamp timestamp  = Timestamp::fromSeconds(0);
+    Ball ball            = Ball(ball_position, ball_velocity, timestamp);
+
+    Robot robot = Robot(0, Point(0, 0), Vector(), Angle::zero(), AngularVelocity::zero(),
+                        timestamp);
+
+    EXPECT_FALSE(Evaluation::robotHasPossession(ball, robot));
+}
+
+TEST(RobotEvaluationTest, has_possession_slightly_off_center_return_true)
+{
+    Point ball_position  = Point(0.07, 0.005);
+    Vector ball_velocity = Vector(0, 0);
+    Timestamp timestamp  = Timestamp::fromSeconds(0);
+    Ball ball            = Ball(ball_position, ball_velocity, timestamp);
+
+    Robot robot = Robot(0, Point(0, 0), Vector(), Angle::zero(), AngularVelocity::zero(),
+                        timestamp);
+
+    EXPECT_TRUE(Evaluation::robotHasPossession(ball, robot));
+}
+
+
+
+TEST(RobotEvaluationTest, has_possession_weird_angle)
```

Comment:
Hmm. Could we rename this to something a bit more descriptive?

Reply:
Changed to "has_possession_robot_on_angle_with_ball_in_dribbler"